app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    def initialize(account, payment_status, user_session = nil)
+      @account         = account
+      @payment_status  = payment_status
+      @gateway_setting = account.gateway_setting
+      @user_session    = user_session
+    end
+
+    def call
+      return if payment_successful?
+
+      limiter = ActionLimiter.new(@account)
+      limiter.perform 'invalid_payment'
+    rescue ActionLimiter::ActionLimitsExceededError
+      account.force_suspend!
```

Comment:
Should we log something here so afterwards we know the reason why the account got suspended?

Reply:
I agree with @guicassolato . Easier to debug later.

Reply:
@guicassolato what about sending an email? And if yes, I send the email for whom? 🤔

Reply:
No need for an email IMHO.

The intention here is to prevent brute force attacks and the new code already takes care of doing whatever one would do in case an email was sent. Besides, it's not like the email message would add anything, as the account data is most likely irrelevant in this case. Finally, sending an email would increase risk of DDoS attack.

Reply:
I ment an email to us/admins to inform that someone is suspended. But I think I'll add it on the logs